packages/babel-core/src/transformation/plugin-pass.ts

```diff
@@ -7,19 +7,32 @@ export default class PluginPass<Options = object> {
   file: File;
   opts: Partial<Options>;
 
-  // The working directory that Babel's programmatic options are loaded
-  // relative to.
+  /**
+   * The working directory that Babel's programmatic options are loaded
+   * relative to.
+   */
   cwd: string;
 
-  // The absolute path of the file being compiled.
+  /** The absolute path of the file being compiled. */
   filename: string | void;
 
-  constructor(file: File, key?: string | null, options?: Options) {
+  /**
+   * Is Babel executed in async mode or not.
+   */
+  isAsync: boolean | undefined;
+
+  constructor(
+    file: File,
+    key?: string | null,
+    options?: Options,
+    isAsync?: boolean,
```

Comment:
i've added `isAsync` as optional argument to the last position, so it shouldn't be a breaking change to anyone who uses `PluginPass` constructor directly. However, i would like to add it as required argument after the `file`.  Let me know if that change is acceptable.

Reply:
We only export `PluginPass` as a type and not as a constructor (see `packages/babel-core/src/index.ts`), so it should be fine to add it as a required property here.

Reply:
Actually, you can just remove the `?`s from the other arguments since they are always always passed. It's probably a leftover from when we converted the repo from Flow to TS.